Foundational update to container-httpd to support external authentication

abellotti opened the pull request:
Foundational update to container-httpd to support external authentication:

- Include RPM's required for external authentication
- Set environment variables used needed by httpd when run under systemd
- Enable and run the message bus with systemd
- Enable and run the httpd service with systemd

Must be merged at same time as: https://github.com/ManageIQ/manageiq-pods/pull/194

abellotti committed:
Foundational update to container-httpd to support external authentication:

- Include RPM's required for external authentication
- Set environment variables used needed by httpd when run under systemd
- Enable and run the message bus with systemd
- Enable and run the httpd service with systemd

abellotti commented:
Marked as WIP:   Depends on https://github.com/ManageIQ/manageiq-pods/pull/194

Fryguy commented on Dockerfile:
```diff
@@ -13,21 +13,69 @@ LABEL name="manageiq-apache" \
+COPY docker-assets/initialize_httpd.sh     /usr/bin
 COPY docker-assets/generate_server_cert.sh /usr/bin
-COPY docker-assets/manageiq.conf /etc/httpd/conf.d/
+COPY docker-assets/manageiq.conf           /etc/httpd/conf.d/
```
This will eventually be a config map right? (Not sure what is *not* included in this PR, since I know you have a bunch coming)

carbonin replied:
> I think this should be changed to something generic, and not MIQ specific, like SERVICE_NAME or CONTAINER_SERVICE_NAME

Probably should also change this name, right?

Fryguy commented on docker-assets/initialize_httpd.sh:
```diff
@@ -0,0 +1,16 @@
+#
+# Container Environment Variables needed
+#
+MANAGEIQ_SERVICE_NAME = "${MANAGEIQ_SERVICE_NAME}"
```
~~I don't understand creating an ENV var from an ENV var~~

EDIT: nvm...I didn't see that you were building a heredoc (I've never seen `!` chars in the heredoc terminator, and it threw me)

Fryguy replied:
I think this should be changed to something generic, and not MIQ specific, like `SERVICE_NAME` or `CONTAINER_SERVICE_NAME`

abellotti replied:
I'm ok on attempting to generalize container-httpd now (not ManageIQ specific), I'll update.

abellotti committed:
Updated container-httpd to be agnostic to ManageIQ and reference it simply as the back-end application.

container-httpd's /etc/httpd/conf.d is now a mapped configmap which
includes application.conf

carbonin commented on docker-assets/application.conf:
```diff
@@ -0,0 +1,53 @@
+# Timeout: The number of seconds before receives and sends time out.
+Timeout 120
```
Are all of these settings really things that *every* app using httpd will want?

At the very least this is making assumptions about ciphers (which people tend to change) and ssl (which I think we are moving to the route and will no longer need in here).

I would think we would provide the entire configuration in our template as a config map, not just parts to be included.

abellotti committed:
Updated container labels in the Dockerfile to replace Apache with httpd
and simplifying the summary and descriptions.
